mlir/lib/Quantum/Transforms/BufferizationPatterns.cpp

```diff
@@ -71,8 +74,21 @@ struct BufferizeSampleOp : public OpConversionPattern<SampleOp> {
         Type tensorType = op.getType(0);
         MemRefType resultType = cast<MemRefType>(getTypeConverter()->convertType(tensorType));
         Location loc = op.getLoc();
-        Value allocVal = rewriter.replaceOpWithNewOp<memref::AllocOp>(op, resultType);
-        rewriter.create<SampleOp>(loc, TypeRange{}, ValueRange{adaptor.getObs(), allocVal},
+
+        // SampleOp's result shape is (shots, num_qubits)
+        // shots might be dynamic, in which case we need to memref alloc from the shots SSA value
+        auto shape = cast<mlir::RankedTensorType>(tensorType).getShape();
+
+        SmallVector<Value> allocSizes;
+        if (shape[0] == ShapedType::kDynamic) {
```

Comment:
Do we really need a memref? I think you can keep the integer, like we can keep floats.

Reply:
Yes, the size to be allocated depends on shots, which is now dynamic, and can't be kept as a int literal attribute

Reply:
Resolving this as stale/unproblematic